rclrs/src/node/graph.rs

```diff
@@ -0,0 +1,568 @@
+use std::collections::HashMap;
+use std::ffi::{CStr, CString};
+use std::slice;
+
+use crate::rcl_bindings::*;
+use crate::{Node, RclrsError, ToResult};
+
+impl Drop for rmw_names_and_types_t {
+    fn drop(&mut self) {
+        // SAFETY: No preconditions for this function
+        unsafe {
+            rcl_names_and_types_fini(self).ok().unwrap();
+        }
+    }
+}
+
+impl Drop for rmw_topic_endpoint_info_array_t {
+    fn drop(&mut self) {
+        // SAFETY: No preconditions for this function
+        unsafe {
+            rmw_topic_endpoint_info_array_fini(self, &mut rcutils_get_default_allocator())
+                .ok()
+                .unwrap();
+        }
+    }
+}
+
+impl Drop for rcutils_string_array_t {
+    fn drop(&mut self) {
+        // SAFETY: No preconditions for this function
+        unsafe {
+            rcutils_string_array_fini(self);
+        }
+    }
+}
+
+/// Topic names and types
```

Comment:
To keep the doc comment style consistent, could you make sure that all doc comments are in the [declarative mood and end with a period](https://github.com/ros2-rust/ros2_rust/blob/main/docs/CONTRIBUTING.md#documentation)?

Reply:
If you want to go the extra mile, you could also add examples and more detailed information. E.g. for this type, we could explain what the key and the value is, and why there might be multiple types per topic.